common/authen.js

```diff
@@ -307,14 +373,86 @@
                 });
             },
 
+            // Makes a request to fetch the most recent session from the server
+            // verifies if that is the same as when the page loaded before calling `cb`
+            validateSessionBeforeMutation: function (cb) {
+                var handleDiffUser = function () {
+                    // modalInstance comes from error modal controller to close the modal after logging in, but before checking to throw an error again
+                    var checkSession = function (modalInstance) {
+                        modalInstance.dismiss("continue");
+                        // check if login state resolved
+                        _getSession().then(function (session) {
+                            if (!session || !_sameSessionAsPrevious) {
+                                handleDiffUser()
+                            } else {
+                                cb();
+                            }
+                        });
+                    }
+
+                    var errorCB = checkSession;
+
+                    if (!_session) {
+                        // for continuing in the modal if there is a user
+                        errorCB = function (modalInstance) {
+                            popupLogin(LOGIN_ERROR_MODAL, function () {
+                                checkSession(modalInstance);
+                            });
+                        }
+                    }
+
+                    throw new Errors.DifferentUserConflictError(_session, _prevSession, errorCB); // cannot dismiss
+                }
+
+                // Checks if an error needs to be thrown because the user is different and continues execution if the login state is resolved
+                // a session must exist before calling this function
+                var validateSessionSubmit = function () {
+                    if (!_sameSessionAsPrevious) {
+                        handleDiffUser();
+                    } else {
+                        // we have a session now and it's the same as when the app started
+                        cb();
+                    }
+                }
+
+                _getSession().then(function (session) {
+                    if (!session) {
+                        var onSuccess = function () {
+                            validateSessionSubmit();
+                        }
+
+                        var onError = function (err) {
+                            // NOTE: user didn't login, what's the error ?
+                            // same error message as duplicate user except "anon"
+                            handleDiffUser();
+                        }
+
+                        // should be modal login
+                        logInHelper(loginWindowCb, "", onSuccess, 'modal', onError, LOGIN_LOGIN_MODAL);
```

Comment:
`LOGIN_LOGIN_MODAL` should be `logService.logActions.LOGIN_LOGIN_MODAL`. Currently, It's throwing a terminal error because of this.

Reply:
Addressed this when you mentioned above that the log actin I defined was unused.